compiler/rustc_middle/src/ty/subst.rs

```diff
@@ -489,31 +494,41 @@ impl<'a, 'tcx> TypeFolder<'tcx> for SubstFolder<'a, 'tcx> {
                     }
                 }
             }
-            _ => r,
+            _ => Ok(r),
         }
     }
 
-    fn fold_ty(&mut self, t: Ty<'tcx>) -> Ty<'tcx> {
+    fn fold_ty(&mut self, t: Ty<'tcx>) -> Result<Ty<'tcx>, Self::Error> {
         if !t.potentially_needs_subst() {
-            return t;
+            return Ok(t);
         }
 
         match *t.kind() {
-            ty::Param(p) => self.ty_for_param(p, t),
+            ty::Param(p) => Ok(self.ty_for_param(p, t)),
             _ => t.super_fold_with(self),
         }
     }
 
-    fn fold_const(&mut self, c: &'tcx ty::Const<'tcx>) -> &'tcx ty::Const<'tcx> {
+    fn fold_const(
+        &mut self,
+        c: &'tcx ty::Const<'tcx>,
+    ) -> Result<&'tcx ty::Const<'tcx>, Self::Error> {
+        if !c.potentially_needs_subst() {
```

Comment:
This is...new. Probably fine, but I'm not sure I'm comfortable embedding such a small change in a PR that's mostly refactoring.

Reply:
Looks like this was erroneously reintroduced from a merge conflict when I rebased #85469.  See https://github.com/rust-lang/rust/pull/85469/files#diff-2990b629b9249f64f215a5763f93342124a492c8aec814f26aa2e71b4324b32bR508-R510